File: python-review-impl/songfactory.py
class SongFactory:
    playlist = {}
    
    @staticmethod
    def get_song(song):
        key = f"{song.song_name}-{song.artist_name}"
        if key not in SongFactory.playlist:
            SongFactory.playlist[key] = song
            print(f"Created new song: {song.song_name}")
        else:
            print(f"Reused existing song: {song.song_name}")
        return SongFactory.playlist[key]

File: src/SongFactory.java
import java.util.HashMap;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class SongFactory {
    static final HashMap<Integer, Song> playList = new HashMap<>();

    public static Song getSong(Song pSong) {
        Integer hash = pSong.getaSongName().hashCode() + pSong.getaArtistName().hashCode();
        if (playList.containsKey(hash)) {
            return playList.get(hash);
        } else {
            playList.put(hash, pSong);
            return playList.get(hash);
        }
    }

    private SongFactory(){}

}
